core/src/main.cpp

```diff
@@ -946,11 +958,17 @@ auto main(int _argc, char* _argv[]) -> int
 		}
 
 		if (vm.count("config-file") == 0) {
-			fmt::print(stderr, "Error: Missing [CONFIG_FILE_PATH] parameter.");
+			fmt::print(stderr, "Error: Missing [CONFIG_FILE_PATH] parameter.\n");
+			return 1;
+		}
+
+		const auto config_file_path = vm["config-file"].as<std::string>();
```

Comment:
Can we get this as a reference?

Reply:
I don't think so?

I've never seen any boost program options example demonstrating acquiring a reference. If you can find one, that would be great.

Reply:
Skimming through the boost filesystem docs, it looks like what you call takes a path ref, and the path ref constructor takes a ref to the string? You can ignore this comment if you don't want to verify, one string copy isn't bad: https://www.boost.org/doc/libs/1_81_0/libs/filesystem/doc/reference.html

Reply:
This is good. Apparently, `.as()` returns a reference to the contents of a `boost::any` object. So I think we're good with just appending an ampersand to `auto`.

Will test that real quick.

Reply:
Confirmed that `fs::path` cannot be used. I did not investigate why. Everything I tried resulted in Boost.Any complaining.

Will continue to use `auto&` and `std::string`.